src/test/codegen/issue-75742-format_without_fmt_args.rs

```diff
@@ -0,0 +1,41 @@
+// min-llvm-version: 10.0.0
+// compile-flags: -C opt-level=3
+#![crate_type = "rlib"]
+
+// Make sure allocation not happen when result of `format!` is unused and
+// there are no formatting arguments.
+
+// CHECK-LABEL: @format_wo_fmt_args
+#[no_mangle]
+pub fn format_wo_fmt_args() {
+    // CHECK-NOT: __rust_alloc
+    format!("a long story");
+}
+
+// CHECK-LABEL: @format_w_empty_fmt_string
+#[no_mangle]
+pub fn format_w_empty_fmt_string() {
+    // CHECK-NOT: __rust_alloc
+    format!("");
```

Comment:
Is this even needed? I think this shouldn't generate any code. I bet the compiler could know this beforehand, unlike having side-effect in https://github.com/rust-lang/rust/issues/75742

Reply:
> Would this generate code if it is not assigned

Nope. Just a single 'ret' instr.

> Would the compiler know this beforehand?

Currently not yet. But in the future MIR could have that ability.